source/DamageProfile.cpp

```diff
@@ -111,11 +111,14 @@ void DamageProfile::PopulateDamage(DamageDealt &damage, const Ship &ship) const
 	double shieldFraction = 0.;
 
 	// Lambda for returning the damage scale that a damage type should
-	// use given the default percentage that is blocked by shields and
-	// the value of its protection attribute.
-	auto ScaleType = [&](double blocked, double protection)
+	// use given the default percentage that is blocked by shields and hull,
+	// and the value of its protection attribute.
+	auto ScaleType = [&](double shieldBlocked, double hullBlocked, double protection)
 	{
-		return damage.scaling * (1. - blocked * shieldFraction) / (1. + protection);
+		return damage.scaling *
+			(((1. - shieldBlocked) * (1. - shieldFraction))
+			+ ((1. - hullBlocked) * (shieldFraction)))
+			/ (1. + protection);
```

Comment:
As best I can tell, this equation does not work at all and should look something more like this.
```suggestion
		double blocked = min(1 - shieldBlocked * shieldFraction, 1 - hullBlocked * (1 - shieldFraction));
		return damage.scaling * blocked / (1. + protection);
```
At least in the cases where shield fraction is 1 or 0 (i.e. your shields entirely block shield damage or your shields are entirely down), this works. We could definitely put extra scrutiny on whether this works for shield fraction values between 0 and 1, though, but from my light messing around in Desmos, this seems to work.

Reply:
This equation mirrors what was already in place, with an implicit "hull never blocks anything" assumption. If a damage type is 50% blocked by shields and 25% blocked by hull, and the target's shields are 50% disrupted, then the damage type should be blocked 37.5%- 1/2 of 50% plus 1/2 of 25%.

What's in the PR now may be using shieldFraction backwards... If shieldFraction is equal to 1 when shields are up and there is no disruption (i.e. "the fraction of the projectile hitting shields" is 1), then my formula is wrong, and the uses of `shieldFraction` and `(1. - shieldFraction)` should be swapped.

Your proposal currently behaves opposite mine when dumped into desmos and only one blockage value is nonzero (which is why I suspect I have `shieldFraction` backwards), but when both are nonzero, the behavior is strange. It's, of course, the minimum of two lines, one pinned to (0, 1), the other pinned to (1, 1) (where, uh, X is shield fraction, Y is % of damage that gets through shields). As an extreme example, if a damage type is said to be 100% blocked by shields and also 100% blocked by hull (useful, I know), your proposal would have a half-disrupted shield allow 50% of the damage to leak through, instead of 0% damage. Because the hull is only half visible, and would allow 50% damage, and the shields are half-visible and would allow 50% damage. Min between 50 and 50 is 50, rather than 0. The use of `min` is _very_ strange to me...

Anyways, I will swap the uses of `shieldFraction` and `(1. - shieldFraction)`, and that should fix the strange behavior you're mentioning.

Reply:
Even when I swapped the shield fraction to what it should have been in your question, it didn't look correct to me. I think it'd be far easier if instead of trying to cram everything into a single equation, we just have a "shield fraction == 1" case (`1 - shieldBlocked`), "shield fraction == 0" case (`1 - hullBlocked`), then a case for when shield fraction is something in between.